### ISSUE TITLE
schemaSync cant't sync schema when a column is both primary key and  foreign key 

### ISSUE THREAD

**Issue by LiJinyao** [typeorm/typeorm#1091]
## Description
Get error when sync same scheme containing a column which is both a primary key and foreign key.

### environment
System: macOS 10.13
database: PostgreSQL 10.0

## Step to reproduce the bug
let's say we have tow entity (I deleted some unnecessary attribute)
CatalogueMap
```typescript
import {BaseEntity, Column, Entity, Index, ManyToOne, OneToOne, PrimaryColumn} from 'typeorm'
import Site from './Site'
@Entity()
export default class CatalogueMap extends BaseEntity {
    // both primary key and foreign key
    @ManyToOne(type => Site, {primary: true})
    site: Site
    @PrimaryColumn()
    originCatalogue: string
}
```
Site
```typescript
import {BaseEntity, Column, Entity, PrimaryColumn} from 'typeorm'
@Entity()
export default class Site extends BaseEntity {
    @PrimaryColumn('varchar', { length: 10 })
    name: string
    @Column()
    url: string
    @Column({nullable: true})
    iconUrl: string
}
```
set `synchronize` to `true` and create connection
```typescript
await createConnection({
        type: 'postgres',
        host: 'localhost',
        port: 5432,
        username: 'postgres',
        password: '',
        database: 'postgres',
        entities: [
            CatalogueMap,
            Site,
        ],
        synchronize: true,
        logging: true
    })
```
in the first time you execute the code, all table will be created normally.

run the code second time, typeorm will check schema and try to sync it, then get the following error， error log is
```log
executing query:  ALTER TABLE "public"."catalogue_map" DROP CONSTRAINT "fk_ad9e375797b7c0306131752a414"
executing query:  ALTER TABLE "public"."catalogue_map" ALTER COLUMN "siteName" DROP NOT NULL
query failed:  ALTER TABLE "public"."catalogue_map" ALTER COLUMN "siteName" DROP NOT NULL

...

(node:7761) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): QueryFailedError: column "siteName" is in a primary key

```

this bug will also happen when you create unique index in the same kind of columns.
```typescript
@Index(['site', 'originCatalogue'], { unique: true })
```

## What is expected

should sync the schema successful in the second time since nothing changed in the entity classes.

## Possible cause

I think the problem is the sync order. since `siteName` is both primary key and foreign key, typeorm try to first sync the foreign key scheme(try to drop and recreate), but because `siteName` also a primary key, so you can't drop the `NOT NULL` constraint. same problem happened when set `@Index(['site', 'originCatalogue'], { unique: true })`, it also prevent you from dropping  `NOT NULL` constraint.

Are there any workaround could bypass this bug or other way to implement same constraint?

**Comment by FrikkieSnyman** [comment]
@pleerock Is there no workaround for this in the meanwhile?

**Comment by AlexMesser** [comment]
This bug is related to `isNullable` detection and synchronization. Currently tracking in #968. 
@FrikkieSnyman current workaround would be adding a `nullable: false` clause to the ManyToOne decorator.

**Comment by pleerock** [comment]
Fixed and will be released in `0.1.8`.
